src/App.tsx

```diff
@@ -15,35 +16,120 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType;
+  isReversed: boolean;
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  let visibleGoods = [...goods];
+
+  if (sortType === SortType.LENGTH) {
+    visibleGoods = [...goods].sort((a, b) => {
+      return a.length - b.length;
+    });
+  } else if (sortType === SortType.ALPHABET) {
+    visibleGoods = [...goods].sort((a, b) => {
+      return a.localeCompare(b);
+    });
+  } else if (sortType === SortType.NONE) {
+    visibleGoods = [...goods];
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isResetButtonVisible, setIsResetButtonVisible] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const reorderedGoods = getReorderedGoods(goodsFromServer, {
+    sortType,
+    isReversed,
+  });
+
+  const sortByAplhabet = () => setSortType(SortType.ALPHABET);
+
+  const sortByLength = () => setSortType(SortType.LENGTH);
+
+  const reset = () => setSortType(SortType.NONE);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${cn({
+            'is-light': sortType !== SortType.ALPHABET,
+          })}`}
+          onClick={() => {
```

Comment:
Not cool :) 

You already have a `sortByAplhabet ` handler, why don't you put all logic in it? 
Basically, that's the idea of handlers.